### PR TITLE
Location picker x y names

### PR DESCRIPTION
**Pull request by peterdudfield**
# Pull Request

## Description

- add options for lat, lon and osgb, so we can use it for Europe data
- add time resolution to nwp
- added to gfs
- added option if the NWP data is per id, or per x,y coords
- options for running inferecne and returning all locations and t0s

Fixes #

## How Has This Been Tested?

CI tests, ran this with PV italy experiements

- [x] Yes

## Checklist:

- [x] My code follows [OCF's coding style guidelines](https://github.com/openclimatefix/nowcasting/blob/main/coding_style.md)
- [ ] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] I have checked my code and corrected any misspellings

### CODE REVIEW

**Comment by peterdudfield** [comment]
Sorry @jacobbieker its a bit big, I'll happy cut this into a few smaller PRs if you like?

File: ocf_datapipes/load/nwp/gfs.py [hunk 1]
```diff
@@ -40,11 +42,29 @@ def open_gfs(zarr_path: Union[Path, str]) -> xr.Dataset:
...
+    _log.info("Loading NWP GFS data")
+
     if "*" in zarr_path:
-        gfs = xr.open_mfdataset(zarr_path, engine="zarr", combine="time", chunks="auto")
+        nwp = xr.open_mfdataset(zarr_path, engine="zarr", combine="time", chunks="auto")
     else:
-        gfs = xr.open_dataset(zarr_path, engine="zarr", mode="r", chunks="auto")
+        nwp = xr.load_dataset(zarr_path, engine="zarr", mode="r", chunks="auto")
+
+    nwp = xr.concat([nwp["dlwrf"], nwp["t"], nwp["u"], nwp["v"], nwp["prate"]], "channel")
```
Comment (peterdudfield) [inline ocf_datapipes/load/nwp/gfs.py:53]: could probably make this more general

Comment (jacobbieker) [reply ocf_datapipes/load/nwp/gfs.py:53]: Yeah, would it be possible to set the variables you want in the config?

File: ocf_datapipes/select/select_t0_time.py [hunk 2]
```diff
@@ -41,15 +44,25 @@ def __iter__(self) -> pd.Timestamp:
 
             if self.return_all_times:
 
+                number_of_locations = next(self.number_locations_datapipe)
```
Comment (jacobbieker) [inline ocf_datapipes/select/select_t0_time.py:47]: If this is None, what happens?

File: ocf_datapipes/utils/consts.py [hunk 3]
```diff
@@ -336,6 +336,9 @@ class BatchKey(Enum):
     "lcc": 47.90454236572895,
     "mcc": 44.22781694449808,
     "hcc": 32.87577371914454,
+    "dlwrf": 319,
```
Comment (jacobbieker) [inline ocf_datapipes/utils/consts.py:339]: I would prefer separating out MetOffice NWP and GFS NWP values, each one will have different values as they are calculated differently.

**Review by jacobbieker — CHANGES_REQUESTED** [review]
Overall looks good to me! Just a few changes possibly? Mostly to keep things a bit more separate, like the constants for GFS vs MetOffice, and in the future ERA5/ECMWF

File: ocf_datapipes/load/nwp/gfs.py (hunk 1 above)
Comment (peterdudfield) [reply ocf_datapipes/load/nwp/gfs.py:53]: yea, for this run, just use all of them, and later add them into config - https://github.com/openclimatefix/ocf_datapipes/issues/97

File: ocf_datapipes/select/select_t0_time.py (hunk 2 above)
Comment (peterdudfield) [reply ocf_datapipes/select/select_t0_time.py:47]: yea, it wont work, ill probably put a more distinct error message

File: ocf_datapipes/utils/consts.py (hunk 3 above)
Comment (peterdudfield) [reply ocf_datapipes/utils/consts.py:339]: thats a good idea

File: ocf_datapipes/select/select_t0_time.py (hunk 2 above)
Comment (jacobbieker) [reply ocf_datapipes/select/select_t0_time.py]: Would you be able to change it so if it is None, then it just does return one location each time? Its how I use it in the MetNet validation for example

Comment (peterdudfield) [reply ocf_datapipes/select/select_t0_time.py:47]: yep, ill do that

**Pull request merged**